src/test/java/peoplesoft/model/job/UniqueJobListTest.java

```diff
@@ -0,0 +1,163 @@
+        assertFalse(uniqueJobList.contains(EATING.getJobId()));
+    }
+
+    @Test
+    public void contains_jobInList_returnsTrue() {
+        uniqueJobList.add(EATING);
+        assertTrue(uniqueJobList.contains(EATING.getJobId()));
+    }
+
+    @Test
+    public void add_nullJob_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueJobList.add(null));
+    }
+
+    @Test
+    public void add_duplicateJob_throwsDuplicateJobException() {
+        Job jobCopy = new JobBuilder(EATING).build();
+        uniqueJobList.add(EATING);
+
+        assertThrows(DuplicateJobException.class, () -> uniqueJobList.add(jobCopy));
+    }
+
+    @Test
+    public void setJob_nullTargetJob_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueJobList.setJob(null, EATING));
+    }
+
+    @Test
+    public void setJob_nullEditedJob_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueJobList.setJob(EATING, null));
+    }
+
+    @Test
+    public void setJob_targetJobNotInList_throwsJobNotFoundException() {
+        assertThrows(JobNotFoundException.class, () -> uniqueJobList.setJob(EATING, EATING));
+    }
+
+    @Test
+    public void setJob_editedJobIsSameJob_success() {
+        uniqueJobList.add(EATING);
+        uniqueJobList.setJob(EATING, EATING);
+        UniqueJobList expectedUniqueJobList = new UniqueJobList();
+        expectedUniqueJobList.add(EATING);
+        assertEquals(expectedUniqueJobList, uniqueJobList);
+    }
+
+    @Test
+    public void setJob_editedJobHasSameIdentity_success() {
+        uniqueJobList.add(EATING);
+        Job jobCopy = new JobBuilder(EATING).build();
+        uniqueJobList.setJob(EATING, jobCopy);
+        UniqueJobList expectedUniqueJobList = new UniqueJobList();
+        expectedUniqueJobList.add(jobCopy);
+        assertEquals(expectedUniqueJobList, uniqueJobList);
+    }
+
+    @Test
+    public void setJob_editedJobHasDifferentIdentity_success() {
+        uniqueJobList.add(EATING);
+        uniqueJobList.setJob(EATING, RUNNING);
+        UniqueJobList expectedUniqueJobList = new UniqueJobList();
+        expectedUniqueJobList.add(RUNNING);
+        assertEquals(expectedUniqueJobList, uniqueJobList);
+    }
+
+    @Test
+    public void setJob_editedJobHasNonUniqueIdentity_throwsDuplicateJobException() {
+        uniqueJobList.add(EATING);
+        uniqueJobList.add(RUNNING);
+        assertThrows(DuplicateJobException.class, () -> uniqueJobList.setJob(EATING, RUNNING));
+    }
+
+    @Test
+    public void remove_nullJob_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueJobList.remove(null));
+    }
+
+    @Test
+    public void remove_jobDoesNotExist_throwsJobNotFoundException() {
+        assertThrows(JobNotFoundException.class, () -> uniqueJobList.remove(EATING));
+    }
+
+    @Test
+    public void remove_existingJob_removesJob() {
```

Comment:
it could be a good idea to check that `add` is actually adding something, but if it doesn't then some other test should fail anyway, so not really a big deal